scripts/ci/backend-tests/run-go-tests-with-profiling.sh

```diff
@@ -0,0 +1,368 @@
+        echo "usage: $0 [-h] -t <tags> [-r <run-pattern>] [-p <parallel>] [-o <output-dir>] [-T <timeout>] [-P <packages>]"
+        echo
+        echo "Options:"
+        echo "  -h: Show this help message."
+        echo "  -t: Go tags (e.g., sqlite, mysql, postgres). Required."
+        echo "  -r: Test run pattern (e.g., '^TestIntegration'). Default: '^TestIntegration'"
+        echo "  -p: Parallelism for main tests. Default: 4"
+        echo "  -o: Output directory for profiling data. Default: profiles"
+        echo "  -T: Timeout for tests. Default: 8m"
+        echo "  -P Package paths to test."
+        echo "      Can be repeated to specify multiple packages."
+        echo "      Can be - to read from stdin (one package per line)."
+        echo "      Default: none (all packages with tests)"
+        echo
+        echo "Environment Variables:"
+        echo "  PROFILE_PACKAGES: Newline-separated list of packages to profile."
+        echo "                    These packages will run in background with profiling enabled."
+        echo "  PROFILE_PARALLEL_RUNS: Newline-separated list of parallel run counts."
+        echo "                         Must have same number of entries as PROFILE_PACKAGES."
+        echo "  MAX_WAIT_MINUTES: Maximum time to wait for profiled tests after main tests (in minutes)."
+        echo "                    Default: 10 minutes"
+        echo
+        echo "Examples:"
+        echo "  # Single package with profiling:"
+        echo "  export PROFILE_PACKAGES=\"pkg/tests/apis/folder\""
+        echo "  export PROFILE_PARALLEL_RUNS=\"3\""
+        echo "  echo \"./pkg/tests/apis/folder\" | $0 -t sqlite -r '^TestIntegration' -P -"
+        echo
+        echo "  # Multiple packages with profiling:"
+        echo "  export PROFILE_PACKAGES=\"pkg/tests/apis/folder"
+        echo "  pkg/tests/apis/dashboard\""
+        echo "  export PROFILE_PARALLEL_RUNS=\"1"
+        echo "  2\""
+        echo "  echo \"./pkg/tests/apis/folder ./pkg/tests/apis/dashboard\" | $0 -t sqlite -r '^TestIntegration' -P -"
+        echo
+        echo "Analyzing Downloaded Artifacts:"
+        echo "  When profiled tests fail, artifacts are generated containing:"
+        echo "    - CPU profiles (cpu_*.prof)"
+        echo "    - Memory profiles (mem_*.prof)"
+        echo "    - Execution traces (trace_*.out)"
+        echo "    - Test logs (test_*.log)"
+        echo "    - Exit codes (exit_*.code)"
+        echo
+        echo "  # Analyze profiles and traces:"
+        echo "  go tool pprof cpu_pkg_tests_apis_dashboard_run1.prof"
+        echo "  go tool pprof -alloc_space mem_pkg_tests_apis_datasource_run1.prof"
+        echo "  go tool trace trace_pkg_tests_apis_datasource_run1.out"
+        echo
+        echo "  For more information:"
+        echo "    - CPU/Memory profiling: https://go.dev/blog/pprof"
+        echo "    - Execution traces: https://pkg.go.dev/cmd/trace https://go.dev/blog/execution-traces-2024"
```

Comment:
Nit: This is common knowledge (easily google-able), we don't need to include that.